### PR TITLE
[14.0][FIX] sh_delivery_shipment: fix list not loaded for shipment

### PR DESCRIPTION
**Pull request by TDu**
Only allow to process lines that are in the allowed picking types for the scenario.

### CODE REVIEW

**Review by jbaudoux — COMMENTED** [review]
It looks like it's already part of `_find_move_lines_domain`

```
     def _find_move_lines_domain(self, shipment_advice):
         """Returns the base domain to look for move lines for a given shipment."""
         return shipment_advice.with_context(
             shipment_picking_type_ids=self.picking_types.ids
         )._find_move_lines_domain()  # Defined in `shipment_advice`
```

File: shopfloor_delivery_shipment/services/delivery_shipment.py [hunk 1]
```diff
@@ -663,7 +663,12 @@ def _find_move_lines_from_product(
     def _find_move_lines_not_loaded_from_shipment(self, shipment_advice):
         """Returns the move lines not loaded at all from the shipment advice."""
         domain = self._find_move_lines_domain(shipment_advice)
-        domain.append(("qty_done", "=", 0))
+        domain.extend(
+            [
+                ("qty_done", "=", 0),
+                ("picking_id.picking_type_id", "in", self.picking_types.ids),
```
Comment (simahawk) [inline shopfloor_delivery_shipment/services/delivery_shipment.py:669]: the picking type filter is already there in `_find_move_lines_domain`, n'est pas?

Comment (simahawk) [reply shopfloor_delivery_shipment/services/delivery_shipment.py:669]: tests?

**Comment by TDu** [comment]
True so I can close this ? @jbaudoux

**Comment by jbaudoux** [comment]
@TDu You can keep it for fixing _find_move_lines_domain as there seems to be the bug.

```
shipment_advice.with_context(
             shipment_picking_type_ids=self.picking_types.ids
         )._find_move_lines_domain()
```

vs

```
shipment_advice._find_move_lines_domain(self.picking_types.ids)
```

File: shopfloor_delivery_shipment/services/delivery_shipment.py (hunk 1 above)
Comment (mmequignon) [reply shopfloor_delivery_shipment/services/delivery_shipment.py:669]: n'est-ce pas*

Comment (mmequignon) [reply shopfloor_delivery_shipment/services/delivery_shipment.py:669]: then it fixes nothing, right?

**Pull request closed without merging**